application/src/main/java/org/togetherjava/tjbot/commands/mathcommands/TeXCommand.java

```diff
@@ -107,6 +97,25 @@ public void onSlashCommand(@NotNull final SlashCommandEvent event) {
             .queue();
     }
 
+    @NotNull
+    private String getLatexToInline(String latex) {
+        Matcher matcher = LATEX_PATTERN.matcher(latex);
+        StringBuilder sb = new StringBuilder();
+        while (matcher.find()) {
+            if (hasMatchedDollarSign(matcher)) {
+                sb.append(matcher.group(1).replace("$", ""));
+            } else {
+                sb.append("\\text{").append(matcher.group(2)).append("}");
+            }
+        }
+        latex = sb.toString();
+        return latex;
+    }
+
+    private boolean hasMatchedDollarSign(Matcher matcher) {
```

Comment:
No need for an extra method. Just create a simple boolean before the `if`. So `boolean hasMatchedDollarSign = matcher.group(1) != null);` is enough.

Reply:
I would also rename it to `isInsideMathRegion`. Cause you essentially have either math-region or text-region.

Its probably easier to understand the code if its looked at like that. In fact, the regex could also be changed to `(%s|%s).formatted(MATH_REGION, TEXT_REGION)` for more readability.